File: 240206/2개씩 그룹짓기/group-of-pairs.py
n = int(input())
arr = list(map(int, input().split()))

# arr 정렬 후 최솟값과 최댓값을 더해가며 remove함

arr.sort()

res = 0
ans = 0

for _ in range(n):
    res = max(arr) + min(arr)
    arr.remove(max(arr))
    arr.remove(min(arr))
    if res >= ans:
        ans = res

print(ans)

File: 240812/2개씩 그룹짓기/group-of-pairs.js
const fs = require("fs");
let input = fs.readFileSync(0).toString().trim().split("\n");
let n = Number(input[0]);
let arr = input[1].trim().split(" ").map(Number);

arr.sort((prev, cur) => prev - cur);

function findMax() {
    let maxVal = arr[0];
    for (let i = 0; i < 2 * n; i++) {
        if (arr[i] + arr[2 * n - i - 1] >= maxVal) {
            maxVal = arr[i] + arr[2 * n - i - 1];
        }
    }
    return maxVal;
}

console.log(findMax());